corehq/apps/reports/static/reports/v2/js/datagrid/data_models.js

```diff
@@ -45,6 +45,11 @@ hqDefine('reports/v2/js/datagrid/data_models', [
                 .css('left', position.left + 'px')
                 .css('top', position.top + "px");
 
+            if (marginTop < 0) {
+                // This makes sure there's no overlap with the header.
+                marginTop = 0;
```

Comment:
Couldn't you do 
``` js
marginTop = Math.min(0, $rows.height() / 2 - 50);
``` 
when you define `marginTop` above?

Reply:
done